src/main/java/frc/robot/intake/IntakeSubsystem.java

```diff
@@ -55,38 +55,70 @@ public double getVolts() {
     }
   }
 
-  /** Creates a new IntakeSubsystem. */
-  public IntakeSubsystem() {}
+  public IntakeState getState() {
+    return state;
+  }
 
-  @Override
-  public void periodic() {
-    io.updateInputs(inputs);
-    Logger.processInputs("Arm", inputs);
+  public IntakeSubsystem(
+      RollerIO rollerIO,
+      PivotIO pivotIO,
+      CANrangeIO leftCanrangeIO,
+      CANrangeIO rightCanrangeIO,
+      String name) {
+    super(rollerIO, pivotIO, name);
+    this.leftCanrangeIO = leftCanrangeIO;
+    this.rightCanrangeIO = rightCanrangeIO;
   }
 
+  @AutoLogOutput(key = "Intake/State")
+  private IntakeState state = IntakeState.IDLE;
+
   public void setState(IntakeState state) {
     this.state = state;
   }
 
-  public void setMotorVoltage(double voltage) {
-    io.setMotorVoltage(voltage);
+  @Override
+  public void periodic() {
+    super.periodic();
+    leftCanrangeIO.updateInputs(leftCanrangeInputs);
+    Logger.processInputs("Intake/leftCanrange", leftCanrangeInputs);
+    rightCanrangeIO.updateInputs(rightCanrangeInputs);
+    Logger.processInputs("Intake/rightanrange", rightCanrangeInputs);
   }
 
-  public void setMotorPosition(Rotation2d targetPosition) {
-    io.setMotorPosition(targetPosition);
+  public double getleftCanrangeDistanceM() {
+    return leftCanrangeInputs.distanceM;
   }
 
-  public boolean isNearAngle(Rotation2d target) {
-    return MathUtil.isNear(target.getDegrees(), inputs.position.getDegrees(), 10.0);
+  public double getRightCanrangeDistanceM() {
+    return rightCanrangeInputs.distanceM;
   }
 
-  // TODO setStateAngleVoltage
-  public Command setStateAngleVoltage() {
-    return Commands.none();
+  @AutoLogOutput(key = "Intake/HasGamePiece")
+  public boolean hasGamePiece() {
+    return getleftCanrangeDistanceM() < 10.0 || getRightCanrangeDistanceM() < 10.0;
```

Comment:
```suggestion
    return getleftCanrangeDistanceM() < 0.05 || getRightCanrangeDistanceM() < 0.05
```

idk we'll tune it. remember this is in meters